### PR TITLE
chore: update logic createNFTCollection

### PR DESCRIPTION
**Pull request by NMINHDAI**
**What does this PR do?**

-   [x] My bugfix

**How to test**

-   Create a detailed description of what you need to do to set this PR up. ie: Does it need migrations? Do you need to install something?
-   Create a step by step list of what the engineer needs to do to test.

**Edge cases**

-   What are the edge cases that you might see in this PR?

**Media (Loom or gif)**

-   Insert media here (if applicable)

### CODE REVIEW

File: pkg/entities/nfts.go [hunk 1]
```diff
@@ -44,6 +44,20 @@ var (
 	}
 )
 
+func (e *Entity) CheckExitNFT(address string) (bool, error) {
```
Comment (trkhoi) [inline pkg/entities/nfts.go:47]: CheckExistNftCollection

File: pkg/entities/nfts_test.go [hunk 2]
```diff
@@ -262,3 +263,120 @@ func TestEntity_GetAllNFTSalesTracker(t *testing.T) {
...
+	defer ctrl.Finish()
+
+	cfg := config.Config{
+		DBUser: "postgres",
+		DBPass: "postgres",
+		DBHost: "localhost",
+		DBPort: "5434",
+		DBName: "mochi_local",
+
+		InDiscordWalletMnemonic: "holiday frequent toy bachelor auto use style result recycle crumble glue blouse",
```
Comment (trkhoi) [inline pkg/entities/nfts_test.go:294]: change all this key config to "sample" or "test"

File: pkg/handler/nfts.go [hunk 3]
```diff
@@ -28,6 +28,14 @@ func (h *Handler) CreateNFTCollection(c *gin.Context) {
 		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 		return
 	}
+
+	checkExitsNFT, _ := h.entities.CheckExitNFT(req.Address)
```
Comment (trkhoi) [inline pkg/handler/nfts.go:32]: CheckExistNftCollection

**Review by trkhoi — CHANGES_REQUESTED** [review]

File: pkg/handler/nfts.go [hunk 4]
```diff
@@ -28,6 +28,14 @@ func (h *Handler) CreateNFTCollection(c *gin.Context) {
 		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 		return
 	}
+
+	checkExitsNFT, _ := h.entities.CheckExistNftCollection(req.Address)
```
Comment (trkhoi) [inline pkg/handler/nfts.go:32]: handle error here

**Review by trkhoi — CHANGES_REQUESTED** [review]

File: pkg/handler/nfts.go [hunk 5]
```diff
@@ -28,6 +28,19 @@ func (h *Handler) CreateNFTCollection(c *gin.Context) {
 		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 		return
 	}
+
+	checkExitsNFT, err := h.entities.CheckExistNftCollection(req.Address)
+	// err == nil means exist
```
Comment (trkhoi) [inline pkg/handler/nfts.go:33]: if err = nil then you duplicate the handle error
what if error is not nil, then ?

File: pkg/handler/nfts.go [hunk 6]
```diff
@@ -28,6 +28,19 @@ func (h *Handler) CreateNFTCollection(c *gin.Context) {
 		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 		return
 	}
+
+	checkExitsNFT, err := h.entities.CheckExistNftCollection(req.Address)
```
Comment (trkhoi) [inline pkg/handler/nfts.go:32]: move this logic behind `checksumAddress, _ := util.ConvertToChecksumAddr(req.Address)`
use the checksumAddress instead of req.Address

**Review by trkhoi — CHANGES_REQUESTED** [review]

File: pkg/handler/nfts.go [hunk 7]
```diff
@@ -28,8 +28,22 @@ func (h *Handler) CreateNFTCollection(c *gin.Context) {
 		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 		return
 	}
+
 	checksumAddress, _ := util.ConvertToChecksumAddr(req.Address)
 
+	checkExitsNFT, err := h.entities.CheckExistNftCollection(checksumAddress)
+	if err != nil {
```
Comment (trkhoi) [inline pkg/handler/nfts.go:35]:
```
if err != nil {
	c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
	return
}
```

**Review by trkhoi — CHANGES_REQUESTED** [review]
